ciao_contrib/_tools/fileio.py

```diff
@@ -306,22 +306,39 @@ def validate_asol(infile):
     although the current check is not guaranteed to be complete,
     just to catch common errors (e.g. using aspect histogram).
 
+    Added in DS10.8.3 is a change from CONTENT=ASPSOL to
+    CONTENT=ASPSOLOBI, so we check we have one of these
+    two here (since we'll switch on this later so may as
+    well error out if it is unexpected).
     """
 
     # Could check things like the block type, but for
     # not just try a simplistic check.
     #
     (keys, cols) = get_keys_cols_from_file(infile)
+    try:
+        content = keys['CONTENT']
+    except KeyError:
+        raise IOError("The file {} is missing the CONTENT keyword.".format(infile))
+
+    if content not in ['ASPSOL', 'ASPSOLOBI']:
+        raise IOError("Aspect solutions have CONTENT=ASPSOL or ASPSOLOBI, but found {} in\n{}".format(content, infile))
```

Comment:
It's more a personal pref, but I'm not a big fan of multi-line error messages (the `\n` before the infile).    Just a pref.